### PR TITLE
[GDR-1975] data.frame-related functions & update package versioning rules

### PR DESCRIPTION
**Pull request by j-smola**
# Description
## What changed?
Related JIRA issue: GDR-1975

## Why was it changed?
- add new rules to lintr (data.frame-related functions)
- update package versioning rules acc to technical meeting

# Checklist for sustainable code base
- [x] I added tests for any code changed/added
- [x] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [x] Package version bumped
- [x] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: tests/testthat/test-linters_config.R [hunk 1]
```diff
@@ -0,0 +1,10 @@
+testthat::test_that("custom `gDR_undesirable_functions` works as expected", {
```
Comment (gladkia) [inline tests/testthat/test-linters_config.R:1]: 👍

File: R/linters_config.R [hunk 2]
```diff
@@ -1,24 +1,22 @@
+gDR_undesirable_functions <-
+  lintr::modify_defaults(
+    defaults = lintr::default_undesirable_functions,
+    "library" = NULL,
+    # we prefer not to use these functions because we use data.table as the primary data format
+    "assert_data_frame",
```
Comment (gladkia) [inline R/linters_config.R:6]: consider some hints if the linter fails, e.g.
`"assert_data_frame" = "please use 'assert_data_table' instead (our primary data format)"`

File: R/linters_config.R [hunk 3]
```diff
@@ -1,24 +1,22 @@
+gDR_undesirable_functions <-
+  lintr::modify_defaults(
+    defaults = lintr::default_undesirable_functions,
+    "library" = NULL,
```
Comment (gladkia) [inline R/linters_config.R:4]: I think the list of exceptions will be longer (e.g. `Sys.setenv, options and mapply`).
Please create a dedicated branch in gDRgenesis and link gDRstyle GDR-1975 to it. Then enable all tests for all packages in ci/packages.yaml (with `test: y` tag). This step will enable gathering the lintr data for all the packages.

File: R/linters_config.R [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
+    # we prefer not to use these functions because we use data.table as the primary data format
+    "assert_data_frame",
+    "rbind.fill"
```
[the remark this answers is not in the document]
Comment (j-smola) [reply R/linters_config.R:7]: Good point.

File: R/linters_config.R (hunk 2 above)
Comment (j-smola) [reply R/linters_config.R:6]: Good point.

File: R/linters_config.R (hunk 3 above)
Comment (j-smola) [reply R/linters_config.R:4]: I have a plan to merge gDRstyle at the end. So, first test it locally on all packages in scope of GDR-1975.
After that I ll check as you proposed.

**Review by gladkia — APPROVED** [review]

**Pull request merged**